service/snap/snap.go

```diff
@@ -536,6 +462,33 @@ func (s *Service) execThenExpect(commandArgs []string, expectation string) error
 	return nil
 }
 
-func (s *Service) runCommand(args ...string) (string, error) {
-	return utils.RunCommand(s.executable, args...)
+func (s Service) runCommand(args ...string) (string, error) {
+	return s.runnable.Execute(s.executable, args...)
+}
+
+func (s Service) runCommandWithRetry(args ...string) (res string, err error) {
+	if resErr := retry.Call(retry.CallArgs{
+		Clock: s.runnable.Clock(),
```

Comment:
I think clocks should be injected via a different route here rather than the runnable instance (see my comment above).